source/Calamari.Terraform/TerraformCliExecutor.cs

```diff
@@ -0,0 +1,212 @@
+
+            commandResult.VerifySuccess();
+
+            return result;
+        }
+
+        public int ExecuteCommand(out string result, params string[] arguments)
+        {
+            var commandResult = ExecuteCommandInternal(ToSpaceSeparated(arguments), out result);
+
+            return commandResult.ExitCode;
+        }
+
+        public int ExecuteCommand(out string result, out CommandResult commandResult, params string[] arguments)
+        {
+            commandResult = ExecuteCommandInternal(ToSpaceSeparated(arguments), out result);
+
+            return commandResult.ExitCode;
+        }
+
+        public void Dispose()
+        {
+            if (AttachLogFile)
+            {
+                if (fileSystem.FileExists(logPath))
+                {
+                    Log.NewOctopusArtifact(fileSystem.GetFullPath(logPath), fileSystem.GetFileName(logPath), fileSystem.GetFileSize(logPath));
+                }
+            }
+        }
+
+        static string ToSpaceSeparated(IEnumerable<string> items)
+        {
+            return string.Join(" ", items.Where(_ => !String.IsNullOrEmpty(_)));
+        }
+
+        CommandResult ExecuteCommandInternal(string arguments, out string result)
+        {
+            var environmentVar = defaultEnvironmentVariables;
+            if (environmentVariables != null)
+            {
+                environmentVar.MergeDictionaries(environmentVariables);
+            }
+
+            var commandLineInvocation = new CommandLineInvocation(TerraformExecutable,
+                arguments, TemplateDirectory, environmentVar);
+
+            var commandOutput = new CaptureOutput();
+            var cmd = new CommandLineRunner(commandOutput);
+            var commandResult = cmd.Execute(commandLineInvocation);
+
+            Log.Info(commandLineInvocation.ToString());
+
+            result = String.Join("\n", commandOutput.Infos);
+
+            return commandResult;
+        }
+
+        void InitialisePlugins()
+        {
+            ExecuteCommandInternal(
+                $"init -no-color -get-plugins={AllowPluginDownloads.ToString().ToLower()} {InitParams}", out _).VerifySuccess();
+        }
+
+        void InitialiseWorkspace()
+        {
+            if (!String.IsNullOrWhiteSpace(Workspace))
+            {
+                ExecuteCommandInternal("workspace list", out var results).VerifySuccess();
+                
+                foreach (var line in results.Split('\n'))
+                {
+                    var workspaceName = line.Trim('*', ' ');
+                    if (workspaceName.Equals(Workspace))
+                    {
+                        ExecuteCommandInternal($"workspace select \"{Workspace}\"", out _).VerifySuccess();
+                        return;
+                    }
+                }
+
+                ExecuteCommandInternal($"workspace new \"{Workspace}\"", out _).VerifySuccess();
+            }
+        }
+
+        class CaptureOutput : ICommandOutput
```

Comment:
Would it be worth just using the existing `CaptureCommandOutput`  class and extracting the information off `Infos`, or are you concerned about the perf\service message parsing?

Reply:
I did look at `CaptureCommandOutput` but that is in the Test project, so it didn't feel like I should bring it in given my requirements were a lot simpler